03/assignment/walog/blog/serializers.py

```diff
@@ -0,0 +1,147 @@
+from rest_framework import serializers
+from django.utils import timezone
+
+from .models import User, Post, Comment, Tag
+from . import views
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['username', 'password']
+
+class TagSerializer(serializers.ModelSerializer):
+    name = serializers.CharField(max_length=200)
+    class Meta:
+        model = Tag
+        fields = ['name']
+
+class CommentSerializer(serializers.ModelSerializer):
+    post = serializers.StringRelatedField()
+    created_by = serializers.StringRelatedField()
+    tags = TagSerializer(many=True)
+
+    class Meta:
+        model = Comment
+        fields = ['id', 'post', 'tags', 'content', 'created_by', 'created_at', 'updated_at', 'is_updated']
+        read_only_fields = ['id', 'updated_at', 'created_at']
+
+    
+    def create(self, validated_data):
+        if 'tags' not in validated_data:
+            return Comment.objects.create(**validated_data)
+
+        tag_data = validated_data.pop('tags')
+        
+        comment = Comment.objects.create(**validated_data)
+        for tag in tag_data:
+            tag = tag['name']
+            tag = tag.strip()
+            if tag is None:
+                continue
+
+            if Tag.objects.filter(name=tag):
+                _tag = Tag.objects.get(name=tag)
+                comment.tags.add(_tag)
+            else:                
+                _tag = Tag(name=tag)
+                _tag.save()
+                comment.tags.add(_tag)
```

Comment:
쿼리가 중복으로 날아가지 않도록 짜는 것이 좋습니다.
```suggestion
            tagObj = Tag.objects.filter(name=tag).first()
            if tagObj is None:
                tagObj = Tag(name=tag)
                tagObj.save()
            comment.tags.add(tagObj)
```